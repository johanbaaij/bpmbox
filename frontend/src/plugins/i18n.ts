import Vue from "vue";
import VueI18n, { LocaleMessages } from "vue-i18n";
import localesEn from "@/locales/en.json";

Vue.use(VueI18n);

const en: LocaleMessages = { en: localesEn };

export default new VueI18n({
  locale: process.env.VUE_APP_I18N_LOCALE || "en",
  fallbackLocale: process.env.VUE_APP_I18N_FALLBACK_LOCALE || "en",
  messages: en
});
