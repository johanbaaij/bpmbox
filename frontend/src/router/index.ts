import Vue from "vue";
import VueRouter from "vue-router";
import Home from "@/views/Home.vue";
import Register from "@/views/Register.vue";
import Login from "@/views/Login.vue";
import FourOhFour from "@/views/404.vue";
import Dashboard from "@/views/Dashboard.vue";
import EventBus from "@/utils/event-bus";

Vue.use(VueRouter);

const routes = [
  {
    path: "/",
    name: "home",
    component: Home
  },
  {
    path: "/404",
    name: "FourOhFour",
    component: FourOhFour
  },
  {
    path: "/register",
    name: "register",
    component: Register,
    meta: { auth: false }
  },
  {
    path: "/login",
    name: "login",
    component: Login,
    meta: { auth: false }
  },
  {
    path: "/dashboard",
    name: "dashboard",
    component: Dashboard,
    meta: { auth: true }
  }
];

const router = new VueRouter({
  mode: "history",
  base: process.env.BASE_URL,
  routes
});

// add notification to "show on next page queue"
// before navigating, add notification to currently

router.beforeEach((_to, _from, next) => {
  next();
  EventBus.$emit("DISPLAY_NEXT_NOTIFICATION");
});

export default router;
