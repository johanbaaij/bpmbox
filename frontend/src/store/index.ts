import Vue from "vue";
import Vuex from "vuex";
import axios from "axios";
import { jsonapiModule } from "jsonapi-vuex";
import tracksModule from "./tracks";
import userModule from "./user";

Vue.use(Vuex);

const api = axios.create({
  baseURL: process.env["VUE_APP_API_URL"],
  headers: {
    "Content-Type": "application/vnd.api+json"
  }
});

const store = new Vuex.Store({
  state: {
    receiving: false
  },
  mutations: {
    receiving(state, payload) {
      state.receiving = payload;
    }
  },
  actions: {
    receiving({ commit }, value) {
      commit("receiving", value);
    }
  },
  modules: {
    tracks: tracksModule,
    user: userModule,
    jv: jsonapiModule(api, {
      preserveJson: true,
      clearOnUpdate: true
    })
  }
});

store.subscribe((_mutation, state: any) => {
  localStorage.setItem("userModule", JSON.stringify(state.user));
});

export default store;
