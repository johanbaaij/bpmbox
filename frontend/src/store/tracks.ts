import { Module } from "vuex";
import TrackQueryParams from "@/TrackQueryParams";

const tracksModule: Module<any, any> = {
  namespaced: true,
  state: {
    serverItemsLength: 0,
    apiLoading: false,
    hideNullBpms: false,
    options: {
      page: 1,
      itemsPerPage: 30,
      sortBy: ["release_title"],
      sortDesc: [false],
      groupBy: [],
      groupDesc: [],
      mustSort: true,
      multiSort: false
    }
  },
  actions: {
    options({ commit, dispatch }, payload) {
      commit("options", payload);
      dispatch("apiData");
    },
    hideNullBpms({ commit, dispatch }, payload) {
      commit("hideNullBpms", payload);
      dispatch("apiData");
    },
    async apiData({ state, rootState, dispatch, commit }) {
      commit("apiLoading", true);
      const filters = {
        hideNullBpms: state.hideNullBpms
      };
      const urlParams = new TrackQueryParams(
        state.options,
        filters
      ).urlParams();
      const jsonapiData = {
        _jv: { type: "track" }
      };

      const response = await dispatch(
        "jv/get",
        [
          jsonapiData,
          {
            url: `/collections/${rootState.user.username}/tracks`,
            params: {
              include: "release",
              ...urlParams
            }
          }
        ],
        { root: true }
      );
      const { _jv } = response;
      commit("serverItemsLength", _jv.json.meta.total);
      commit("apiLoading", false);
    }
  },
  mutations: {
    options(state, payload) {
      state.options = payload;
    },
    serverItemsLength(state, payload) {
      state.serverItemsLength = payload;
    },
    apiLoading(state, payload) {
      state.apiLoading = payload;
    },
    hideNullBpms(state, payload) {
      state.hideNullBpms = payload;
    }
  }
};

export default tracksModule;
