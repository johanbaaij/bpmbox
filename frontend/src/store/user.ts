import { Module } from "vuex";

const userModule: Module<any, any> = {
  namespaced: true,
  state: {
    username: null
  },
  actions: {
    username({ commit }, payload) {
      commit("username", payload);
    }
  },
  mutations: {
    username(state, payload) {
      state.username = payload;
    },
    initialise(state) {
      if (localStorage.getItem("userModule")) {
        const userStorage = JSON.parse(
          localStorage.getItem("userModule") as string
        );
        state = Object.assign(state, userStorage);
      }
    }
  }
};

export default userModule;
