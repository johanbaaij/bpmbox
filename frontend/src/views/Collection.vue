<template>
  <v-container fluid>
    <v-row>
      <v-col>
        <v-tabs>
          <v-tab to="tracks">Tracks</v-tab>
        </v-tabs>
      </v-col>
    </v-row>
    <router-view />
  </v-container>
</template>

<script lang="ts">
import { Component, Vue, Prop } from "vue-property-decorator";
import MetaInfo from "vue-meta";

@Component({
  name: "collection",
  metaInfo(): MetaInfo {
    return {
      title: this.$i18n.t("Collection.title") as string
    };
  }
})
export default class Collection extends Vue {
  $cable: any;

  beforeMount() {
    if (this.$store.state.user.username === null) {
      // UnhandledPromiseRejectionWarning
      this.$router.replace({ name: "home" });
    }
  }

  get username() {
    return this.$store.state.user.username;
  }

  mounted() {
    this.$cable.subscribe({
      channel: "CollectionsChannel",
      username: this.username
    });
  }
}
</script>

<style lang="sass"></style>
