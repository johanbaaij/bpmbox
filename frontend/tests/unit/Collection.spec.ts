import "./setup";

// Dependencies
import Vuetify from "vuetify";
import i18n from "@/plugins/i18n";
import router from "@/router";
import store from "@/store";

// Components
import Collection from "../../src/views/Collection.vue";

// Utilities
import { mount, createLocalVue } from "@vue/test-utils";

const localVue = createLocalVue();

describe("Collection.vue", () => {
  let vuetify: any;

  beforeEach(() => {
    vuetify = new Vuetify();
  });

  it("should match snapshot", () => {
    const wrapper = mount(Collection, {
      localVue,
      router,
      vuetify,
      i18n,
      store
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
