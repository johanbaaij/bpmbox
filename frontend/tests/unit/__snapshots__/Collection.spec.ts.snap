// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collection.vue should match snapshot 1`] = `
<div class="container container--fluid">
  <div class="row">
    <div class="col">
      <div class="v-tabs theme--light">
        <div role="tablist" class="v-item-group theme--light v-slide-group v-tabs-bar v-tabs-bar--is-mobile primary--text">
          <div class="v-slide-group__prev v-slide-group__prev--disabled">
            <!---->
          </div>
          <div class="v-slide-group__wrapper">
            <div class="v-slide-group__content v-tabs-bar__content">
              <div class="v-tabs-slider-wrapper" style="transition: none;">
                <div class="v-tabs-slider"></div>
              </div><a href="/tracks" class="v-tab" tabindex="0" aria-selected="true" role="tab">Tracks</a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class="container container--fluid">
    <div class="row justify-center">
      <div class="col col-auto">
        <h1 class="text-center">
          bpmbox
        </h1>
        <h2 class="text-center">
          find BPMs for your Discogs collection
        </h2>
      </div>
    </div>
    <div class="row justify-center">
      <div class="col-sm-8 col-md-6 col-lg-4 col"><span><div class="v-card v-sheet theme--light"><div class="v-card__text pb-0"><p>Enter your Discogs username to get started.</p> <span><div class="v-input theme--light v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-text-field--placeholder"><div class="v-input__control"><div class="v-input__slot"><div class="v-input__prepend-inner"><div class="v-input__icon v-input__icon--prepend-inner"><i aria-hidden="true" class="v-icon notranslate mdi mdi-account theme--light"></i></div></div><div class="v-text-field__slot"><input id="input-19" placeholder="johanbaaij" type="text"></div></div><div class="v-text-field__details"><div class="v-messages theme--light"><div class="v-messages__wrapper"></div></div></div></div></div></span></div>
      <!---->
      <div class="v-card__actions">
        <div class="spacer"></div> <button type="button" class="v-btn v-btn--contained theme--light v-size--default primary"><span class="v-btn__content">Find BPMs</span></button>
      </div>
    </div></span>
  </div>
</div>
</div>
</div>
`;
